packages/mongodb-rag-core/src/MongoDbEmbeddedContentStore.ts

```diff
@@ -80,40 +80,43 @@ export function makeMongoDbEmbeddedContentStore({
       });
     },
 
+    /**
+      @param vector - The vector to search for nearest neighbors to.
+      @param options - Options for performing a nearest-neighbor search.
+     */
     async findNearestNeighbors(vector, options) {
       const {
         indexName,
         path,
         k,
         minScore,
         filter,
+        numCandidates,
       }: Partial<FindNearestNeighborsOptions> = {
         // Default options
-        indexName: "default",
+        indexName: "vector_index",
         path: "embedding",
         k: 3,
         minScore: 0.9,
-
         // User options override
         ...(options ?? {}),
       };
       return embeddedContentCollection
         .aggregate<WithScore<EmbeddedContent>>([
           {
-            $search: {
+            $vectorSearch: {
               index: indexName,
-              knnBeta: {
-                vector,
-                path,
-                k,
-                filter,
-              },
+              queryVector: vector,
+              path,
+              limit: k,
+              numCandidates: numCandidates ?? k * 15,
```

Comment:
From [`$vectorSearch` docs](https://www.mongodb.com/docs/atlas/atlas-vector-search/vector-search-stage/):
> we recommend a ratio of ten to twenty nearest neighbors

Just checking, did you choose 15 to split the difference here or is there other guidance that points at 15?

Reply:
> did you choose 15 to split the difference here

exactly this

---

and user can override this default behavior by providing `numCandidates`